core/src/main/java/org/mule/runtime/core/internal/execution/ResolvePreviousTransactionInterceptor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.execution;
+
+import org.mule.runtime.core.api.execution.ExecutionCallback;
+import org.mule.runtime.core.api.transaction.Transaction;
+import org.mule.runtime.core.api.transaction.TransactionConfig;
+import org.mule.runtime.api.tx.TransactionException;
+import org.mule.runtime.core.api.transaction.TransactionCoordination;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ResolvePreviousTransactionInterceptor<T> implements ExecutionInterceptor<T> {
```

Comment:
add some kind of comment saying that this is for compatibility only. Moving to a compatibility package would also rock